### DISCUSSION TITLE
Any way for file send?

### DISCUSSION THREAD

**Discussion by s-safak** [jchristn/SuperSimpleTcp#74] [Q&A]
This is the really good and easy to use component. But I would like to send a small size file with this component. Is there any way to use StreamWriter and StreamReader? 

Especially StreamReader reader = new StreamReader(tcpClient.GetStream());   I couldn't find a solution for this line? Any suggestion pls ?

**Comment by jchristn** [comment]
Hi @s-safak the library doesn't itself provide a way to send a file.  You can send a message to the recipient to let them know how large the file will be, then use ```SendAsync``` within a ```using``` block for your ```FileStream```, i.e.
```
long len = new FileInfo("./mydoc.txt").Length;
using (FileStream fs = new FileStream("./mydoc.txt"))
{
  await tcpClient.SendAsync(len, fs);
}
```
